plugins/catalog/src/js/repositories/components/AddRepositoryFormModal.js

```diff
@@ -79,14 +79,18 @@ class AddRepositoryFormModal extends React.Component {
   }
 
   getButtonDefinition() {
+    const { pendingRequest } = this.props;
+
     return [
       {
         text: i18nMark("Cancel"),
         className: "button button-primary-link flush-left",
         isClose: true
       },
       {
-        text: i18nMark("Add Repository"),
+        text: pendingRequest
+          ? i18nMark("Adding...")
```

Comment:
This didn't actually work for me. Where does the pendingRequest prop come from?

Reply:
Good catch. It should be coming from the result of `addRepository$`, but we're only ever setting it to false, and we don't pass it as a prop to the `AddRepositoryFormModal` component